src/NodaTime.Test/TimeIntervalTest.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2011 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using System;
+using NodaTime.Text;
+using NUnit.Framework;
+
+namespace NodaTime.Test
+{
+    public class TimeIntervalTest
+    {
+        private static readonly LocalTime SampleStart = new LocalTime(10, 00);
+        private static readonly LocalTime SampleEnd = new LocalTime(17, 00);
+        
+        private static readonly LocalTime SampleStartInverted = new LocalTime(23, 00);
+        private static readonly LocalTime SampleEndInverted = new LocalTime(1, 00);
+
+        [Test]
+        public void Duration_True()
+        {
+            var interval = new TimeInterval(SampleStart, SampleEnd);
+            Assert.AreEqual(Duration.FromHours(7), interval.Duration);
+        }
+        [Test]
+        public void Period_True()
+        {
+            var interval = new TimeInterval(SampleStart, SampleEnd);
+            Assert.AreEqual(Period.FromHours(7), interval.Period);
+        }
+        
+        [Test]
+        public void InvertedTimeInterval_Duration_True()
+        {
+            var interval = new TimeInterval(SampleStartInverted, SampleEndInverted);
+            Assert.AreEqual(Duration.FromHours(2), interval.Duration);
+        }
+        
+        [Test]
+        public void InvertedTimeInterval_Period_True()
+        {
+            var interval = new TimeInterval(SampleStartInverted, SampleEndInverted);
+            Assert.AreEqual(Period.FromHours(2), interval.Period);
+        }
+        
+        [Test]
+        public void InvertedTimeInterval_Contains_True()
```

Comment:
Please could you convert all the tests that are data-driven to use TestCase? For example, I'd expect:

```csharp
[Test]
[TestCase(23, 0, true)]
... // Other true cases
[TestCase(1, 0, false)]
... // Other false cases
public void Contains_InvertedTimeInterval(int hour, int minute, bool expectedResult)
```

Reply:
Done. Please tell if I should change some other tests.